pdf/pdfselection.cpp

```diff
@@ -267,51 +267,78 @@ void PDFSelection::Private::textBoxAtPoint(const QPointF &point, Position positi
         }
         break;
     }
-    case PDFSelection::Private::Before: {
+    case PDFSelection::Private::Before:
         // Find the last box index in pageIndex that is before @point,
         // including at @point. If none is found in this page, returns
         // the last box of the previous page.
-        QPointF reducedCoordPoint {point.x() / at.second.width(),
-                (point.y() - at.second.y()) / at.second.height()};
-        for (PDFDocument::TextList::const_iterator box = boxes.begin();
-             box != boxes.end(); box++) {
-            /* Stop counting boxes up as soon as a box after reducedCoordPoint is found. */
-            if (box->first.y() > reducedCoordPoint.y() ||
-                (box->first.y() + box->first.height() > reducedCoordPoint.y() &&
-                 box->first.x() > reducedCoordPoint.x())) {
-                if (*boxIndex == -1) {
-                    *pageIndex -= 1;
-                    if (*pageIndex >= 0)
-                        *boxIndex = canvas->document()->textBoxesAtPage(*pageIndex).length() - 1;
-                }
-                return;
-            }
-            *boxIndex += 1;
-        }
-        *boxIndex = boxes.length() - 1;
-        break;
-    }
     case PDFSelection::Private::After: {
         // Find the first box index in pageIndex that is after @point,
         // including at @point. If none is found in this page, returns
         // the first box of the next page.
         QPointF reducedCoordPoint {point.x() / at.second.width(),
                 (point.y() - at.second.y()) / at.second.height()};
-        *boxIndex = 0;
+        
+        int index = 0;
+        PDFSelection::Private::Position previousBoxPosition = PDFSelection::Private::Before;
+        PDFSelection::Private::Position currentBoxPosition;
```

Comment:
Cosmetics, but since used only inside for loop, could limit scope by introducing variable there.